smslibrary/src/main/java/com/eis/smslibrary/SMSMessageParser.java

```diff
@@ -0,0 +1,124 @@
+package com.eis.smslibrary;
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+
+import com.eis.communication.MessageHandler;
+
+import it.lucacrema.preferences.PreferencesManager;
+
+/**
+ * Singleton class used to parse String to SMSMessage and back
+ * Uses a strategy to parse messages, so that any user can update it to its preferred parser
+ * By defaults it uses a default strategy, defined by the library
+ *
+ * @author Luca Crema, Alberto Ursino, Marco Mariotto
+ */
+public class SMSMessageParser implements MessageHandler<String, String, SMSMessage> {
+
+    private static final String PREFERENCE_DEFAULT_STRATEGY = "";
+
+    private SMSParseStrategy parseStrategy;
+    private static SMSMessageParser instance;
+
+    /**
+     * Private constructor
+     * @param context The calling context
+     */
+    private SMSMessageParser(Context context) {
```

Comment:
I would also check if the context is null here, throwing an exception, just in case.